### DISCUSSION TITLE
How to identify the forked repo in a pull request workflow ?

### DISCUSSION THREAD

**Discussion by lelegard** [community/community#38609] [Actions] [answered]
Hi,

When a workflow is run on a `pull_request`, how can we identify the forked repo from which the pull request was originated?

Let me explain the use case.
- A project is managed in repo `org/project`
- There is a test suite in a sibling repo `org/project-test`, in the same organisation. The test suite is large and in a separate repo to avoid cluttering the main repo.
- In `org/project`, a CI workflow is triggered on `push` and `pull_request`
- After building the project, the workflow downloads a snapshot of the test suite from `org/project-test` and runs the tests.

This works fine for push and most pull requests. However, when a user submits a pull request from `user/project` which requires a modification of the test suite, the checks of the pull request fails because the test suite in `org/project-test` is no longer appropriate. In practice, the user also forked `org/project-test` as `user/project-test` and submits the corresponding PR on the test suite too.

However, at the time the PR is sent from `user/project`, the `org/project-test` is not appropriate for the test. I would like to modify the workflow so that when it is run in a pull request, we get the origin of the PR (`user/project`), check if a repo `user/project-test` exists (from the same user) and download it instead of `org/project-test` to run the test suite.

However, I haven't found a way in the workflow to identity the forked repo which is the origin of the PR, here `user/project`.

How can I get this information in the workflow?

**Comment by ossanna16** [comment]
Hi there @lelegard and welcome to our community! Thank you for asking a great question 🙂
While you are waiting for a reply to your question, here are a few things you can do:

To get started, introduce yourself in our [official introduction thread](https://github.com/orgs/community/discussions/23204)
To connect with other new users, check out this Discussion to [Meet Your GitHub Community Team](https://github.com/orgs/community/discussions/28776), further your GitHub knowledge by working through our [GitHub Skill 🆙 exercises](https://github.com/orgs/community/discussions/32121), and participate in our [weekly Fun Friday chats](https://github.com/orgs/community/discussions/30912)

**Comment by lelegard** [comment, marked as answer]
In the absence of response, I have found the solution here:
~~~
    env:
      ORIGIN_REPO: ${{ github.event.pull_request.head.repo.full_name || github.repository }}
~~~
In the case of a PR, it contains the `user/project` from which the PR was triggered. In all other cases (direct push, manual dispatch), it contains the `org/project`, name of the current repo.

In similar cases, looking for the reference of an information, the recipe is `echo '${{ toJSON(github) }}'` in a dummy action. It displays all the hierarchy of information in the workflow log. That was the information I originally missed.
